test/test_optim.py

```diff
@@ -556,16 +557,22 @@ def test_peak_memory_foreach(self, device, dtype, optim_info):
             self.assertLessEqual(mt_max_mem, st_max_mem + intermediate_size * nintermediates)
 
 
-    @onlyCUDA
+    @onlyNativeDeviceTypes
     @optims([optim for optim in optim_db if "fused" in optim.supported_impls], dtypes=[torch.float64])
     def test_fused_matches_forloop(self, device, dtype, optim_info):
+        if device == 'cpu' and optim_info.optim_cls.__name__ != "Adam":
+            # For cpu, we only support fused with Adam now
```

Comment:
Is the plan to add all the fusion counterparts of CUDA? If so, add a TODO here?

Reply:
Yes, TODO added.